docs/manual/src/internals/bindings_render_pipeline.md

```diff
@@ -0,0 +1,307 @@
+* See [Lifting and Lowering](./lifting_and_lowering.md) for details on `UniFfiTag`.
+
+When a UniFFI bindgen command runs, it reads/deserializes the exported metadata symbols from the Rust library.
+Then the metadata is passed on to the next stage of the pipeline
+See `uniffi_bingen::macro_metadata` and `uniffi_meta::reader` for details.
+
+Here's how the metadata might look for the above function:
+```
+FnMetadata {
+    module_path: "arithmetic",
+    name: "add",
+    is_async: false,
+    inputs: [
+        FnParamMetadata {
+            name: "a",
+            ty: UInt64,
+            by_ref: false,
+            optional: false,
+            default: None,
+        },
+        FnParamMetadata {
+            name: "b",
+            ty: UInt64,
+            by_ref: false,
+            optional: false,
+            default: None,
+        },
+    ],
+    return_type: Some(
+        UInt64,
+    ),
+    throws: Some(
+        Enum {
+            module_path: "arithmetic",
+            name: "ArithmeticError",
+        },
+    ),
+    checksum: Some(
+        39863,
+    ),
+    docstring: None,
+},
+```
+
+
+## Metadata from the UDL file
+
+UniFFI also generates metadata from UDL files and merges this with the proc-macro metadata.
+
+## Metadata -> BindingsIr
+
+This phase transforms the metadata into a `BindingsIr`, an intermediate representation of the generated bindings.
+It adds derived information about the FFI, for example:
+
+* The FFI scaffolding function for each Rust function
+* The FFI scaffolding functions to clone/free object references
+* The FFI type that each type is lowered into/lifted from
+* If a type is used as an error or not
+
+When the function metadata is transformed into the `BindingsIr`, [lowered FFI types and FFI scaffolding
+function](./lifting_and_lowering.md) are added.
+
+```
+Function {
+    name: "add",
+    ...
+    arguments: [
+        Argument {
+            name: "a",
+            ty: Type {
+                kind: UInt64,
+                ffi_type: FfiType::UInt64,
+                ...
+            },
+            ...
+        },
+        ...
+    ],
+    return_type: ReturnType {
+        ty: Some(
+            Type {
+                kind: UInt64,
+                ffi_type: FfiType::UInt64,
+                ...
+            },
+        ),
+        ...
+    },
+    throws_type: ThrowsType {
+        ty: Some(
+            Type {
+                kind: Enum {
+                    module_path: "arithmeticpm",
+                    name: "ArithmeticError",
+                },
+                ffi_type: FfiType::RustBuffer,
+                ...
+            },
+        ),
+        ...
+    },
+    ffi_func: FfiFunctionRef {
+        name: "uniffi_arithmeticpm_fn_func_add",
+        ...
+    },
+    ...
+}
+```
+
+## Specializing the BindingsIr
+
+This phase transforms the general `BindingsIr` into a language-specific one.  Each bindings generate
```

Comment:
I think, "Each bindings generate . . . " should probably be, "Each binding generated . . . "